tests/integration/utils.py

```diff
@@ -463,3 +472,170 @@ def _validate_expected_dep_file_contents(dep_contents_file: Path, output_dir: Pa
         dep_file = output_dir / "deps" / path
         assert dep_file.exists()
         assert dep_file.read_text() == expected_content
+
+
+def retrieve_changed_files_from_git() -> tuple[Path, ...]:
+    repo = Repo(".", search_parent_directories=True)
+    # >>> type(repo.branches)
+    # <class 'git.util.IterableList'>
+    # Despite the fact stated above mypy does not believe one can use 'in' with
+    # repo.branches because branches is an alias to heads and heads are decorated
+    # with @property:
+    #  Unsupported right operand type for in ("Callable[[], IterableList[Head]]")
+    main = "main" if "main" in repo.branches else "origin/main"  # type: ignore
+    try:
+        files = repo.git.diff("--name-only", f"{main}..HEAD").split("\n")
```

Comment:
So, locally, this only works for committed changes. Do we want to add a check for non-committed changes?

Reply:
Yes, but that's IMO a good candidate for a follow-up, this is doing enough already for both the local environments where you can test your changes right before submitting them to the gating CI to crunch for many minutes as well as for the gating to crunch a bit less :).
I'm saying ^this because I already have another follow-up in mind - rebase detection - and those 2 could be paired in another PR.